Dockerfile

```
@@ -32,12 +30,12 @@ RUN apt-get update -y && \
       libtool \
       make \
       openssl \
-      python3.8 \
-      python3.8-dev \
+      python3.12 \
+      python3.12-dev \
       python3-pip \
       unzip \
       vim-tiny && \
-    pip install --no-cache-dir --upgrade pip 'setuptools<71' && \
+    pip install --no-cache-dir --upgrade 'setuptools<71' && \
```

```suggestion
    pip install --no-cache-dir --upgrade setuptools && \
```
With Ubuntu 24.04 this should not be needed anymore, could you please double check also in the other PRs?